lib/modules/datasource/index.ts

```diff
@@ -209,7 +209,9 @@ function resolveRegistryUrls(
         'Custom registries are not allowed for this datasource and will be ignored'
       );
     }
-    return datasource.defaultRegistryUrls ?? [];
+    return is.function_(datasource.defaultRegistryUrls)
+      ? datasource.defaultRegistryUrls()
+      : datasource.defaultRegistryUrls;
```

Comment:
Is it okay to return an empty array to satisfy strict null checks
```js
return is.nullOrUndefined(datasource.defaultRegistryUrls)
      ? []
      : is.function_(datasource.defaultRegistryUrls)
      ? datasource.defaultRegistryUrls()
      : datasource.defaultRegistryUrls;
  ```

Reply:
Could maybe be achieved with ?? []

Reply:
```js
return is.function_(datasource.defaultRegistryUrls)
      ? datasource.defaultRegistryUrls() ?? []
      : datasource.defaultRegistryUrls ?? [];
      ```

Reply:
Is the first one really required?

Reply:
Yes it is an optional property.

Reply:
My bad. The first one isn't actually needed.